aima-core/src/test/java/aima/test/core/unit/search/framework/MetricsTest.class

```diff
@@ -0,0 +1,47 @@
+package aima.test.core.unit.search.framework;
+
+import aima.core.search.framework.Metrics;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+
+public class MetricsTest {
+
+    private Metrics metrics;
+
+    @Before
+    public void before() {
+        metrics = new Metrics();
+    }
+
+    @Test
+    public void testGetInt() throws Exception {
+        int x = 893597823;
+        metrics.set("abcd",x);
+        assertEquals(123, metrics.getInt("abcd"));
+    }
```

Comment:
This is a good start. However, I would recommend restructuring your tests. Instead of having a 'throws Exception' and having a JUnit assert check fail (you really want all the assertions to pass), I would test for the positive and negative cases with the different assert APIs. For example. the above could be written as:

@Test
public void testGetInt() {
     int x = 893597823;
     metrics.set("abcd",x);
     assertNotEquals(123, metrics.getInt("abcd"));
     assertEquals(x, metrics.getInt("abcd"));
}